### ISSUE TITLE
[React] Login with signInWithRedirect does not work with option 4

### ISSUE THREAD

**Issue by sumitsahoo** [firebase/firebase-js-sdk#8598]
### Operating System

macOS 15.0.1

### Environment (if applicable)

Chrome 130, Edge 130, Firefox, 

### Firebase SDK Version

10.14.1 & 11.0.1

### Firebase SDK Product(s)

Auth

### Project Tooling

React app with webpack

### Detailed Problem Description

As per the Firebase [documentation](https://firebase.google.com/docs/auth/web/redirect-best-practices) on self hosted helper files for signInWithRedirect, we have added the files to proper directories i.e. `/__/auth/` and `/__/firebase/` as seen below. We have added to the `public` directory as a build process is involved for the `React` app.

<img width="199" alt="Screenshot 2024-10-25 at 3 21 34 PM" src="https://github.com/user-attachments/assets/29b6537d-cf00-4001-92b7-def89ae43a4c">

For the deployed version to work and fix files being downloaded without being processed we have to update `nginx` config.

<img width="624" alt="Screenshot 2024-10-25 at 3 24 33 PM" src="https://github.com/user-attachments/assets/341123af-80bf-4968-b58b-4cb26fad9c62">

But it never works, the redirect does happen but the result inside `getRedirectResult` is always `null`

```
// Handle redirect and get user details
    const getUserDetails = async () => {
      try {
        const userCred = await getRedirectResult(auth);
        console.log("userCred", userCred);
      }
      catch (error) {
        console.log("error", error);
      }
    };
```

To test it locally we used `mkcert` to enable HTTPS but still, the result was the same. Steps to enable HTTPS are below:

1. Install mkcert
```
npm install -g mkcert
```
2. Create CA

```
mkcert create-ca
```
3. Create Cert

```
mkcert create-cert
```
4. Install the certificate and trust it in the system (macOS & Windows)
5. Update the run script in `package.json`

```
HTTPS=true SSL_CRT_FILE=cert.crt SSL_KEY_FILE=cert.key react-scripts start
```
To make the redirect work, we had to rename `handler` to `handler.html` and `iframe` to `iframe.html`. However, there are no clear instructions on how to test this login locally from Firebase, which I find disappointing.

With these also the result is the same. This is such a big issue and I want to know why there is no traction on this from Firebase. This is a dealbreaker for the Enterprise use case. Related issue: #8329 

### Steps and code to reproduce issue

Below is the code for initiating the `signInWithRedirect`

```
const auth = getAuth(app);
const result = await signInWithRedirect(auth, provider);
```

And in App.js we are observing the redirect response 

```
// Handle redirect and get user details
    const getUserDetails = async () => {
      try {
        const userCred = await getRedirectResult(auth);
        console.log("userCred", userCred);
      }
      catch (error) {
        console.log("error", error);
      }
    };
```

The result is always null.

**Comment by jbalidiong** [comment]
Hi @sumitsahoo, thank you for reaching out to us and apologies for the delayed response. Upon checking, we would be needing additional information to check if you've properly implemented the Option 4:

Using the [self host helper code step](https://firebase.google.com/docs/auth/web/redirect-best-practices#self-host-helper-code), could you share the link of the following:

```
<your-domain>/__/auth/handler
<your-domain>/__/auth/handler.js
<your-domain>/__/auth/experiments.js
<your-domain>/__/auth/iframe
<your-domain>/__/auth/iframe.js
<your-domain>/__/firebase/init.json 
```

We would like to check if the contents in those files are the same as your `https://<project>.firebaseapp.com/__/auth/handler`, etc. Also make sure that you've done step 3 update authorized `redirect_uri` and your `authDomain`.

Please file a ticket via our [support channel](https://firebase.google.com/support) and provide the needed information to help us further investigate the issue.  You can also include this Github issue link for reference.

We are currently tracking this internally in b/375588284.

**Comment by sumitsahoo** [comment]
@jbalidiong Sure I will share the details in the support ticket. I had created one from GCP but it got closed.

**Comment by sumitsahoo** [comment]
@jbalidiong I forgot to mention that `firebase/init.json` is not downloadable from `https://<project>.firebaseapp.com/__/firebase/init.json` since they do not exist. We had to create it manually and give below details:

```
{
    "apiKey": "<key>",
    "authDomain": "<custom_domain>"
}
```

The documentation needs to be updated since the file is no longer maintained at app url.

**Comment by jbalidiong** [comment]
Thanks for the additional information, @sumitsahoo. Our engineers are currently checking on this. As additional help, can you configure nginx to accept `HTTPS/SSL` requests? Here is a [guide](https://www.techrepublic.com/article/how-to-enable-ssl-on-nginx/) on how to enable ssl on nginx.

**Comment by sumitsahoo** [comment]
I have tried many things but locally it does not work, no matter what I try. I had to add conditional approach to use `popup` when running locally and use `redirect` on deployed version. Though I am sad to see very less support from Google and Firebase team on this.

Updated `Nginx` config here:

<img width="600" alt="Screenshot 2024-11-21 at 12 33 45 PM" src="https://github.com/user-attachments/assets/03f8a729-18f0-467c-804a-41945bd3b740">

Sample conditional check here (could also uses `await`):

<img width="546" alt="Screenshot 2024-11-21 at 12 38 02 PM" src="https://github.com/user-attachments/assets/df152d86-587f-4712-8f40-082392764329">
